### PR TITLE
Fix jquery::framework to use WebAsset, and add "defer=false" attribute

### PR DESCRIPTION
**Pull request by Fedik**
This is fix/update for https://github.com/joomla/joomla-cms/pull/22460

Because `jquery::freamwork` not always in use after WebAsset was introduced.

### CODE REVIEW

File: build/build-modules-js/settings.json [hunk 1]
```diff
@@ -258,7 +258,12 @@
         "provideAssets": [
           {
             "name": null,
-            "js": ["jquery.min.js"]
+            "js": ["jquery.min.js"],
```
Comment (laoneo) [inline build/build-modules-js/settings.json:261]: Should we not do these kind of changes un its own pr in the main repo?

Comment (Fedik) [reply build/build-modules-js/settings.json:261]: that true, I even have plan to do it for whole `jquery.php`
but I have include it here, because without this your pull not work

Comment (Fedik) [reply build/build-modules-js/settings.json:261]: ah, no, sorry, I confused with `settings.json`
`settings.json` changes must be here, because we going to add attribute (defer:fase) to `jquery.min.js`, which already exist in `settings.json`

**Comment by Fedik** [comment]
you can ignore changes for `jquery::freamwork`, I will do PR for whole jquery.php
but you need changes from `settings.json` to make your pull work after https://github.com/joomla/joomla-cms/pull/22435 was merged

**Comment by laoneo** [comment]
Actually the pr in the main repo dies work. What ia broken there?

**Comment by Fedik** [comment]
@laoneo just check how it work on latest 4.0-dev, you will see a looot "jQuery undefined" in browser console :wink: 
especially when enable Debug, because a DebugBar depends from jQuery

and undefined cssVars()

**Comment by laoneo** [comment]
Ah wait, sorry. I was in a total different pr #23062.
